### PR TITLE
State api deprecation: Update samples to default to InMemoryStore and…

### PR DESCRIPTION
**Pull request by carlosscastro**
… simplify moving to Azure

+Add reference to Microsoft.Bot.Builder.Azure in samples
+Register azure module and InMemoryStore in samples
+Adjust comments to simplify using table storage or docdb in samples

### CODE REVIEW

File: CSharp/Samples/AlarmBot/Global.asax.cs [hunk 1]
```diff
@@ -12,52 +15,42 @@ public class Global : System.Web.HttpApplication
...
         {
-            {
-                // http://docs.autofac.org/en/latest/integration/webapi.html#quick-start
-                var builder = new ContainerBuilder();
-
-                // register the Bot Builder module
-                builder.RegisterModule(new DialogModule());
-                // register the alarm dependencies
-                builder.RegisterModule(new AlarmModule());
-
```
Comment (willportnoy) [inline CSharp/Samples/AlarmBot/Global.asax.cs]: these are still important

File: CSharp/Samples/AlarmBot/Microsoft.Bot.Sample.AlarmBot.csproj [hunk 2]
```diff
@@ -101,19 +87,82 @@
       <HintPath>..\..\packages\Autofac.WebApi2.3.4.0\lib\net45\Autofac.Integration.WebApi.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Chronic, Version=0.3.2.0, Culture=neutral, PublicKeyToken=3bd1f1ef638b0d3c, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\Chronic.Signed.0.3.2\lib\net40\Chronic.dll</HintPath>
+    </Reference>
+    <Reference Include="EntityFramework, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089, processorArchitecture=MSIL">
```
Comment (willportnoy) [inline CSharp/Samples/AlarmBot/Microsoft.Bot.Sample.AlarmBot.csproj]: >EntityFramework [](start = 24, length = 15)

big dependency for the in-memory store?

File: CSharp/Samples/AlarmBot/Global.asax.cs (hunk 1 above)
Comment (carlosscastro) [reply CSharp/Samples/AlarmBot/Global.asax.cs]: good catch!!! after 50 samples my brain stopped working

**Comment by JasonSowers** [comment]
@carlosscastro what should we do about the conflicts?

**Pull request merged**